metals/src/main/scala/scala/meta/internal/metals/codeactions/OrganizeImports.scala

```diff
@@ -131,21 +127,23 @@ class OrganizeImportsQuickFix(
 
   override val kind: String = OrganizeImportsQuickFix.kind
   override protected val title: String = OrganizeImportsQuickFix.title
-  override protected def isCallAllowed(
+  override protected def disabledReason(
       file: AbsolutePath,
       params: CodeActionParams,
-  ): Boolean = {
+  ): Option[String] = {
     val hasUnused = params
       .getContext()
       .getDiagnostics()
       .asScala
       .collect { case ScalacDiagnostic.UnusedImport(name) => name }
       .nonEmpty
 
-    if (hasUnused && !diagnostics.hasDiagnosticError(file)) {
-      true
+    if (!hasUnused) {
```

Comment:
This should arguably also check `if (!isScalaOrSbt(file))`, as the other one does?

Reply:
That would make sense, though I don't think Metals will be invoked for anything else than Scala and Java files. Maybe we should be explicit about not doing organize import on Java